### ISSUE TITLE
Make HTTP and UWSGI ports configurable or move to random high ports to avoid conflicts on VPN Containers

### ISSUE THREAD

**Issue by agrider** [bbilly1/tubearchivist#103]
On trying to deploy this application, I found that it was failing to start. I found the below logs during startup. The logs declare that uWSGI is failing to bind to an address 8080, even though the doc shows the app running on 8000. It looks like the default HTTPS port. At the very least, the container should expect that the undocumented port might be unavailable on the host machine and document it, if not explicitly expose it. 

More generally, it would be helpful if there were ENV variables provided to configure the HTTP and HTTPS ports Django intends to use inside the container, as 8000 and 8080 are extremely common ports with a high likelihood of conflict. While port remapping on the outside of the container is often able to work around this, privacy minded folks tend to use VPN containers and route their networking through them (`network_mode: "service:vpn_container"`). In this case, it's not possible to use port remapping in docker, we need to control the port being run *inside* the container, so it would be helpful to be allowed to reconfigure the port. Otherwise, we can only run one container attached to the VPN which uses those very common ports internally.

```
*** Starting uWSGI 2.0.20 (64bit) on [Fri Dec 10 14:43:38 2021] ***
compiled with version: 10.2.1 20210110 on 27 November 2021 12:48:51
os: Linux-5.11.0-41-generic #45~20.04.1-Ubuntu SMP Wed Nov 10 10:20:10 UTC 2021
nodename: 388e7310cdf1
machine: x86_64
clock source: unix
detected number of CPU cores: 8
current working directory: /app
writing pidfile to /tmp/project-master.pid
detected binary path: /usr/local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
probably another instance of uWSGI is running on the same address (:8080).
bind(): Address already in use [core/socket.c line 769]
VACUUM: pidfile removed.
```

**Comment by bbilly1** [comment]
I think there seems to be a misunderstanding. `8080` is used container internally only for the uwsgi socket for the django application, take a look at the [uwsgi.ini](https://github.com/bbilly1/tubearchivist/blob/master/uwsgi.ini) file. But that's on localhost of the container's internal network, that doesn't need to be exposed or accessed from outside of the container.  
Then nginx serves the interface and listens on `8000` and will make a uswgi_pass to `localhost:8080` for the application and has the needed aliases to serve the user content, e.g. video files and artwork. That's defined in [nginx.conf](https://github.com/bbilly1/tubearchivist/blob/master/nginx.conf). 

So I'm not sure why this fails in your setup, complete docker-compose and startup logs might give more insights as this is most likely not related to Tube Archivist. As a matter of fact, I just run a quick testing nginx webserver on `8080`. That does not interfere with Tube Archivist's `8000` and both can coexist in peace. 

But yes, I wasn't aware of a scenario where docker's port redirecting like `9000:8000` to access the interface on port `9000` would be insufficient.

**Comment by agrider** [comment]
Gotcha, yeah that's my bad. I just threw out a random assumption as far as what `8080` was being used for, as I didn't have time to do any debugging of my own yesterday. You are right it doesn't need exposed for anyone outside the container. It does consume a common port on the container though. This won't have any impact on those who don't use another container's networking, but it can cause issues for those that do use another container for networking like a VPN container. Here's an [article describing the scenario](https://www.naturalborncoder.com/linux/2021/02/19/making-a-docker-container-use-a-vpn/), effectively the two containers are treated as one for networking and therefore cannot bind the same port. It's a pretty common scenario for apps like this where folks are using the app to archive online comment that they pass requests through a VPN container, which itself often is using these common ports. That's all I was trying to get at, it would be good to avoid using port `8080` if possible. For now, I and anyone else experiencing this issue can override these manually by mounting our own `uwsgi.ini` and `nginx.conf`. Ideally, it would be good to either use a random high port that is unlikely to have conflicts or make the uwsgi port configurable on the container easily like via env var.

Sorry for the confusion and thanks for the great app!

**Comment by bbilly1** [comment]
TIL, nginx doesn't know environment variables. Very disappointing. Don't really know if that is a good solution, but passing `TA_PORT` or `TA_UWSGI_PORT` variables will change the config files with `sed` at startup. Ugly but it works...

I'll keep looking for a better way to do that. But yes, mounting your own conf files is not a future proof solution. :-)

**Comment by agrider** [comment]
Ah yeah, it looks like it can't directly leverage the ENV vars, but you can use something like [envsubstr](https://developpaper.com/how-to-use-environment-variables-in-nginx-configuration-files/), though that's really just a slightly cleaner version of what you are doing with sed in the end. Either way, thanks for the enhancement, it's much appreciated!

**Comment by bbilly1** [comment]
Yes, I've seen `envsubstr` as well, but this is doing basically the same thing as `sed`: It rewrites the conf file, there is nothing dynamic about this as you'd expect with an environment variable. 

This is now merged in v0.0.9.
